library/ssl_tls.c

```diff
@@ -855,8 +890,10 @@ int mbedtls_ssl_derive_keys( mbedtls_ssl_context *ssl )
     defined(MBEDTLS_SSL_PROTO_TLS1_2)
     if( ssl->minor_ver >= MBEDTLS_SSL_MINOR_VERSION_1 )
     {
-        mbedtls_md_hmac_starts( &transform->md_ctx_enc, mac_enc, mac_key_len );
-        mbedtls_md_hmac_starts( &transform->md_ctx_dec, mac_dec, mac_key_len );
+        if( ( ret = mbedtls_md_hmac_starts( &transform->md_ctx_enc, mac_enc, mac_key_len ) ) != 0 )              
```

Comment:
I've investigated the test failure that led to #1437. Earlier in `mbedtls_ssl_derive_keys`, `transform->md_ctx_enc` and `transform->md_ctx_dec` are set up in ciphersuites that use an HMAC, and are left in their null state in ciphersuites that use AEAD.

Historically, all ciphersuites used HMAC. When AEAD ciphersuites were added, this part of the code should have been changed to something like
```
if( ciphersuite_uses_hmac )
{
    mbedtls_md_hmac_starts( &transform->md_ctx_enc, mac_enc, mac_key_len );
    mbedtls_md_hmac_starts( &transform->md_ctx_dec, mac_dec, mac_key_len );
}
```
plus the error checking that this PR introduces.

In the current code structure, a way to test `ciphersuite_uses_hmac` is to test `mac_key_len != 0`.

Rather than make a separate PR, it would be simpler to fix this problem in this PR.